app/src/main/java/eu/kanade/tachiyomi/data/backup/restore/restorers/MangaRestorer.kt

```diff
@@ -295,8 +297,11 @@ class MangaRestorer(
         categories: List<Long>,
         backupCategories: List<BackupCategory>,
     ) {
-        val dbCategories = getCategories.await()
-        val dbCategoriesByName = dbCategories.associateBy { it.name }
+        if (this.dbCategoriesMapByNameCache == null) {
```

Comment:
Having this inline with a force non-null is really messy. Put it behind a function which fills the cache on first call instead. then this reduces to something like
```kotlin
val dbCategoriesByName = dbCategoriesByName()
```
Even though it is still effectively a side effect, it makes it more clear that this is just a cache fill. I would have suggested turning it into a delegated property, but suspend makes it difficult.

Reply:
I have moved this up to `BackupRestorer`